hazelcast/include/hazelcast/client/imap.h

```diff
@@ -37,1132 +37,1381 @@
 #include "hazelcast/client/protocol/codec/codecs.h"
 #include "hazelcast/client/spi/ClientContext.h"
 
-#if  defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64)
+#if defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64)
 #pragma warning(push)
-#pragma warning(disable: 4251) //for dll export
+#pragma warning(disable : 4251) // for dll export
 #endif
 
 namespace hazelcast {
-    namespace client {
-        class hazelcast_client;
+namespace client {
+class hazelcast_client;
 
-        namespace spi {
```

Comment:
Do we really want namepsace and class definition alignments all to left with no indentation? By the way, I was hoping that you auto-generated the namespace closing comment (as you put in the PR description) but I can not see them here.

Reply:
> Do we really want namepsace and class definition alignments all to left with no indentation? 

I think, yes. Because most code in a file generally lives in the same namespace, and adding an indent for every namespace is wasting valuable space on the left side of the code. This makes lines long and hard to read.

> By the way, I was hoping that you auto-generated the namespace closing comment (as you put in the PR description) but I can not see them here.

If the namespace's content is only one line it doesn't generate a closing comment, which is the case for this particular line. There is a new parameter in clang-format 14 called ShortNamespaceLines which allows you to control exactly this but it's very new an might not available on everywhere yet - as a matter of fact LLVM 14 was released today -. Anyway, I think this should not be important, it won't put comments if there is only one line.